qiskit/circuit/quantumcircuitdata.py

```diff
@@ -14,131 +14,15 @@
-    qubits: Tuple[Qubit, ...]
-    """A sequence of the qubits that the operation is applied to."""
-    clbits: Tuple[Clbit, ...]
-    """A sequence of the classical bits that this operation reads from or writes to."""
-
-    def __init__(
-        self,
-        operation: Operation,
-        qubits: Iterable[Qubit] = (),
-        clbits: Iterable[Clbit] = (),
-    ):
-        self.operation = operation
-        self.qubits = tuple(qubits)
-        self.clbits = tuple(clbits)
-
-    def copy(self) -> "CircuitInstruction":
-        """Return a shallow copy of the :class:`CircuitInstruction`."""
-        return self.__class__(
-            operation=self.operation,
-            qubits=self.qubits,
-            clbits=self.clbits,
-        )
-
-    def replace(
-        self,
-        operation: Optional[Operation] = None,
-        qubits: Optional[Iterable[Qubit]] = None,
-        clbits: Optional[Iterable[Clbit]] = None,
-    ) -> "CircuitInstruction":
-        """Return a new :class:`CircuitInstruction` with the given fields replaced."""
-        return self.__class__(
-            operation=self.operation if operation is None else operation,
-            qubits=self.qubits if qubits is None else qubits,
-            clbits=self.clbits if clbits is None else clbits,
-        )
-
-    def __repr__(self):
-        return (
-            f"{type(self).__name__}("
-            f"operation={self.operation!r}"
-            f", qubits={self.qubits!r}"
-            f", clbits={self.clbits!r}"
-            ")"
-        )
-
-    def __eq__(self, other):
-        if isinstance(other, type(self)):
-            # Ordered from fastest comparisons to slowest.
-            return (
-                self.clbits == other.clbits
-                and self.qubits == other.qubits
-                and self.operation == other.operation
-            )
-        if isinstance(other, tuple):
-            return self._legacy_format() == other
-        return NotImplemented
 
-    # Legacy tuple-like interface support.
-    #
-    # For a best attempt at API compatibility during the transition to using this new class, we need
-    # the interface to behave exactly like the old 3-tuple `(inst, qargs, cargs)` if it's treated
-    # like that via unpacking or similar.  That means that the `parameters` field is completely
-    # absent, and the qubits and clbits must be converted to lists.
 
-    def _legacy_format(self):
-        # The qubits and clbits were generally stored as lists in the old format, and various
-        # places assume that they will certainly be lists.
-        return (self.operation, list(self.qubits), list(self.clbits))
-
-    def __getitem__(self, key):
-        return self._legacy_format()[key]
-
-    def __iter__(self):
-        return iter(self._legacy_format())
-
-    def __len__(self):
-        return 3
+CircuitInstruction = qiskit._accelerate.quantum_circuit.CircuitInstruction
 
 
 class QuantumCircuitData(MutableSequence):
```

Comment:
We _really_ ought to get rid of this class lol.

Reply:
Heh, yeah. We'll axe it in a separate PR soon I'm sure 😄.